lib/auth/auth.go

```diff
@@ -2446,56 +2454,57 @@ func (a *Server) mfaAuthChallenge(ctx context.Context, user string, u2fStorage u
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	var enableTOTP, enableU2F bool
+	var enableTOTP, enableU2F, enableWebauthn bool
 	switch apref.GetSecondFactor() {
 	case constants.SecondFactorOTP:
-		enableTOTP, enableU2F = true, false
+		enableTOTP = true
 	case constants.SecondFactorU2F:
-		enableTOTP, enableU2F = false, true
+		enableU2F = true
+	case constants.SecondFactorWebauthn:
+		enableWebauthn = true
+	case constants.SecondFactorOn, constants.SecondFactorOptional:
+		enableTOTP, enableU2F, enableWebauthn = true, true, true
 	default:
-		// Other AuthPreference types don't restrict us to a single MFA type,
-		// send all challenges.
-		enableTOTP, enableU2F = true, true
+		return nil, trace.Errorf("unexpected second_factor value: %s", apref.GetSecondFactor())
```

Comment:
Would this be an appropriate place to use `trace.BadParameter`?

Reply:
Yes, we apparently default to `trace.BadParameter`. I'll fix it.